pioneers-testnet/pioneers-exercise-2.md

```diff
@@ -0,0 +1,170 @@
+
+| Id#Index     | This identifies the UTxO that is the source of the Ada – you should get this from  myaddr. |
+|--------------|--------------------------------------------------------------------------------------------|
+| Out+lovelace | Hex encoded address that will receive the Ada and the amount to send in Lovelace.          |
+
+You will also need to give it a time to live, in slots (ttl) and a fee (in lovelace).  Use the following settings:
+
+| ttl 	| 1000    	|
+|-----	|---------	|
+| fee 	| 1000000 	|
+
+The settings that are used here indicate that the transaction should be processed within 1,000 slots and  that it will cost no more than 1 Ada to submit (a safe value for a simple transaction on the Testnet).  You must pay this (small) fee every time you successfully process a transaction on the Cardano Blockchain.  It will be distributed as part of the pool rewards.  The source of this fee is encoded in the transaction as a UTxO.  We are now ready to sign the transaction and submit it to the chain.
+
+4. Generate a signing key, txsign.
+
+```bash
+$ cardano-cli keygen …
+```
+
+Sign your transaction in txbody using the signing key that you have just produced:
+
+```bash
+$ cardano-cli shelley transaction sign \
+ 	--tx-body-file txbody \
+ 	--signing-key-file txsign \
+ 	--network-magic … \
+ 	--tx-body-file txbody
+```
+
+You will need to give the correct Network Magic Id for the Testnet, as supplied by IOHK in the Genesis file (e.g. 42).
+
+5. Confirm that your node is synced, and then submit your transaction to the Blockchain:
+
+```bash
+$ cardano-cli shelley transaction submit \
+ 	--network-magic … \
+ 	--tx-body-file txbody
+```
+
+You should receive a confirmation that the transaction has been submitted to your local instance of the node, but it may not yet be recorded on the blockchain.  If the node is not running or it cannot be contacted, you will see an error.  Just kill/restart the node in this case and try again.
+
+6. After 2 minutes (possibly earlier), your Ada should be transferred to your new address.
+
+```bash
+ 	$ cardano-cli shelley query filtered-utxo \
+ 	 	--address … \
+ 	 	--network-magic …
+```
+
+7. Finally, build a transaction that sends a total of 1,000 Ada to two different addresses (a multi-address transaction).   Generate two new address keys, and build a transaction that sends 250 Ada to the first address, and 750 Ada to the second address.  Record the details of these addresses so that you can use them again later.
+
+```bash
+$ cardano-cli shelley address key-gen …
+$ cardano-cli shelley address key-gen …
+$ cardano-node shelley transaction build-raw …
+```
+
+The required fee will be higher than before, since part of the cost is based on the number of addresses that the output is sent to.  You can check that the fee will be sufficient before you build the transaction using:
+
+```bash
+$ cardano-node shelley transaction calculate-min-fee …
```

Comment:
```suggestion
$ cardano-cli shelley query protocol-parameters --testnet-magic 42 > protocol-parameters.json
$ cardano-node shelley transaction calculate-min-fee --protocol-params-file protocol-parameters.json …
```

This command requires a json file with protocol informations, maybe will be more helpful.